### PR TITLE
Search

### PR DESCRIPTION
**Pull request by isaack87**
Search Bar Implementation along with form Box css

### CODE REVIEW

File: client/src/components/questionsAnswers/MainAnswerQuestionBox.jsx [hunk 1]
```diff
@@ -16,47 +16,50 @@ const MainAnswerQuestionBox = (props) => {
...
+            <QuestionReport qid={data.qID} />
+            <AnswerForm pid={props.productId} qid={data.qID} />
+          </p>
         </ul>
       </div>
-
       <div>
         <div>
           <ul>
             {data.answers.slice(0, props.visibleAnswers).map(answerlist => (
```
Comment (heltan) [inline client/src/components/questionsAnswers/MainAnswerQuestionBox.jsx:37]: I like this use of higher order functions! very nice

File: client/src/components/questionsAnswers/MainAnswerQuestionBox.jsx [hunk 2]
```diff
@@ -16,47 +16,50 @@ const MainAnswerQuestionBox = (props) => {
...
-              </p>
+                <p className='answers-asked'> A: {answerlist.body.toString().toLowerCase()}
+                  <br />
+                  <br />
+                  <span className="user-data">
+                    { answerlist.answerer_name === 'Seller'
+                      ? <b> {answerlist.answerer_name} </b>
+                      : <normal>{answerlist.answerer_name}</normal> }
+                    __
+                    <Moment format="MMMM-DD-YYYY" date={answerlist.date} />
```
Comment (Louis-La) [inline client/src/components/questionsAnswers/MainAnswerQuestionBox.jsx:47]: I like the separation of components.

File: client/src/components/questionsAnswers/AnswerForm.jsx [hunk 3]
```diff
@@ -82,40 +82,51 @@ class AnswerForm extends React.Component {
...
+          </label>
+
+          <label>
+            Enter Email*
+            <input value={this.state.email}
+              onChange={this.onChangeEmail}
+              name="email"
+              type="text"
+              placeholder="Why did you like the product or not**"
+              pattern="/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/"
```
Comment (heltan) [inline client/src/components/questionsAnswers/AnswerForm.jsx:110]: this is really interesting and i can't wait to see how you use this

File: client/src/components/questionsAnswers/QuestionForm.jsx [hunk 4]
```diff
@@ -42,62 +47,109 @@ class QuestionForm extends React.Component {
...
-          <br />
-
-          <input type="submit"
-            className="a-btn-submit"
+      <div className="form-box">
+        <form>
+        <h1>Ask Your Question <button type="submit" className='X' onClick={this.onClose}>X</button></h1>
+          <label>
+            Enter UserName*
+            <input value={this.state.name}
```
Comment (Louis-La) [inline client/src/components/questionsAnswers/QuestionForm.jsx:118]: I'm assuming it's hardcoding data for now

File: client/src/components/questionsAnswers/QuestionAnswerState.jsx [hunk 5]
```diff
@@ -12,44 +12,27 @@ class QuestionsAnswersState extends React.Component {
...
-      qid: [],
-      searchTerm: "",
       productId: props.productId,
     };
-    this.search = this.search.bind(this);
+
     this.sendProductIdToServer = this.sendProductIdToServer.bind(this);
     this.getQuestions = this.getQuestions.bind(this);
     this.loadMoreAnswers = this.loadMoreAnswers.bind(this);
     this.loadMoreQuestions = this.loadMoreQuestions.bind(this);
```
Comment (heltan) [inline client/src/components/questionsAnswers/QuestionAnswerState.jsx:26]: the function names are clear and easy to understand their function, good job

**Review by Louis-La — COMMENTED** [review]
Code looks good.  No conflict

**Review by heltan — APPROVED** [review]

**Review by Louis-La — APPROVED** [review]
Approved

**Review by Louis-La — APPROVED** [review]
Approved

**Pull request closed without merging**